libclamav/7z_iface.c

```diff
@@ -189,18 +184,19 @@ int cli_7unz(cli_ctx *ctx, size_t offset)
                     break;
 
                 cli_dbgmsg("cli_7unz: Saving to %s\n", tmp_name);
-                if (cli_writen(fd, outBuffer + offset, outSizeProcessed) != outSizeProcessed)
+                if (cli_writen(fd, outBuffer + offset, outSizeProcessed) != outSizeProcessed) {
                     found = CL_EWRITE;
-                else if ((found = cli_magic_scan_desc(fd, tmp_name, ctx, name)) == CL_VIRUS)
-                    viruses_found++;
+                }
+
+                found = cli_magic_scan_desc(fd, tmp_name, ctx, name);
```

Comment:
It looks like we may still want an `else {}` around this. I don't know why I removed it.

Reply:
Under further consideration, I think this is so we can at least scan partial writes... I don't know that it really matters.  I guess I'll keep this change.